core/templates/pages/topic-editor-page/questions-tab/topic-questions-tab.component.ts

```diff
@@ -16,93 +16,114 @@
+import { CategorizedSkills, TopicsAndSkillsDashboardBackendApiService } from 'domain/topics_and_skills_dashboard/topics-and-skills-dashboard-backend-api.service';
 import { Subscription } from 'rxjs';
+import { QuestionsListService } from 'services/questions-list.service';
+import { FocusManagerService } from 'services/stateful/focus-manager.service';
+import { TopicEditorStateService } from '../services/topic-editor-state.service';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { SkillSummary } from 'domain/skill/skill-summary.model';
+import { Question } from 'domain/question/QuestionObjectFactory';
+import { Misconception } from 'domain/skill/MisconceptionObjectFactory';
 
-angular.module('oppia').component('questionsTab', {
-  template: require('./topic-questions-tab.component.html'),
-  controller: [
-    '$scope', '$window', 'FocusManagerService', 'QuestionsListService',
-    'TopicEditorStateService', 'TopicsAndSkillsDashboardBackendApiService',
-    function(
-        $scope, $window, FocusManagerService, QuestionsListService,
-        TopicEditorStateService, TopicsAndSkillsDashboardBackendApiService) {
-      var ctrl = this;
-      ctrl.directiveSubscriptions = new Subscription();
-      $scope.getSkillsCategorizedByTopics = null;
-
-      var _initTab = function() {
-        $scope.question = null;
-        $scope.skillId = null;
-        $scope.topic = TopicEditorStateService.getTopic();
-        $scope.topicRights = TopicEditorStateService.getTopicRights();
-        $scope.groupedSkillSummaries = (
-          TopicEditorStateService.getGroupedSkillSummaries());
-        $scope.skillIdToRubricsObject =
-          TopicEditorStateService.getSkillIdToRubricsObject();
-        $scope.allSkillSummaries = [];
-        $scope.allSkillSummaries = $scope.allSkillSummaries.concat(
-          $scope.topic.getUncategorizedSkillSummaries());
-        for (var i = 0; i < $scope.topic.getSubtopics().length; i++) {
-          var subtopic = $scope.topic.getSubtopics()[i];
-          $scope.allSkillSummaries = $scope.allSkillSummaries.concat(
-            subtopic.getSkillSummaries());
-        }
-        TopicsAndSkillsDashboardBackendApiService.fetchDashboardDataAsync()
-          .then(function(response) {
-            $scope.getSkillsCategorizedByTopics = (
-              response.categorizedSkillsDict);
-            $scope.getUntriagedSkillSummaries = (
-              response.untriagedSkillSummaries);
-          });
-        $scope.canEditQuestion = $scope.topicRights.canEditTopic();
-        $scope.misconceptions = [];
-        $scope.questionIsBeingUpdated = false;
-        $scope.questionIsBeingSaved = false;
-        $scope.emptyMisconceptionsList = [];
-      };
+@Component({
+  selector: 'oppia-topic-questions-tab',
+  templateUrl: './topic-questions-tab.component.html'
+})
+export class TopicQuestionsTabComponent implements OnInit,
+ AfterViewInit, OnDestroy {
+  question!: Question;
```

Comment:
Could you please point out where this variable is being used? I couldn't find it anywhere in this component's class or its template.

Reply:
Removed redundant variables